### PR TITLE
Fix negative overflow and missing '..' on struct lit base exprs

### PR DESCRIPTION
**Pull request by Kha**

### CODE REVIEW

File: tests/target/expr.rs [hunk 1]
```diff
@@ -183,3 +183,9 @@ fn arrays() {
 
     [1 + 3, 4, 5, 6, 7, 7, fncall::<Vec<_>>(3 - 1)]
 }
+
+fn struct_exprs() {
+    Foo { a: 1, b: f(2) };
+    Foo { a: 1, b: f(2), ..g(3) };
+    LoooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooongStruct { ..base };
```
Comment (marcusklaas) [inline tests/target/expr.rs]: I think we should add a line break here. Could you add a FIXME with a reference to https://github.com/nrc/rustfmt/issues/231?

**Comment by marcusklaas** [comment]
Looks good! With added FIXME, r=me.

**Comment by Kha** [comment]
Rebased

**Comment by Kha** [comment]
I also looked into why the rewrite of `base` fails in the first place, and I think the fields' width should actually be `v_budget` instead of `h_budget` in the block-style multi-line case, but of course we don't know that up front with `ListTactic::HorizontalVertical`, so I'm at a loss of how to implement this :) .

**Comment by marcusklaas** [comment]
Thanks!

**Comment by marcusklaas** [comment]
I think we may split off the horizontal/ vertical decision logic to a separate function, at some point.

**Pull request merged**